playbooks/hc-ansible-deployment/tasks/luks_tang_setup.yml

```diff
@@ -11,23 +11,20 @@
   pre_tasks:
     # Check if root device is encrypted
     - name: Find whether root device is a luks device
-      command: cryptsetup isLuks {{ rootdevice }}
+      shell: cryptsetup isLuks {{ rootdevice }}
       register: result
-      when: when: gluster_infra_tangservers is defined
-      run_once: true
-
-    - name: Get the results of isLuks command
-      command: echo $?
-      register: result
-      when: when: gluster_infra_tangservers is defined
+      ignore_errors: True
+      when: gluster_infra_tangservers is defined
       run_once: true
+      tags:
+        - bindtang
 
     - name: Check if root device is encrypted
-      assert:
-        that:
-          - "result != 0"
-        fail_msg: "The given {{ rootdevice }} is not encrypted."
-      when: when: gluster_infra_tangservers is defined
+      fail:
+        msg: "The given boot device {{ rootdevice }} is not encrypted."
+      when: "result.failed == true"
```

Comment:
Apologies for my ignorance, but in line 15 we register a variable result that contain the output of shell command which was executed at line 14, and in line 25 we check if the output of shell command i.e, result.failed == true,  but in and instance what if the  result contains "Device <device-name> does not exist or access denied" , will this condition handle it?

Reply:
If the device does not exist then also the commad will fail and in error the reason will be there.

Reply:
oh cool i missed that